### ISSUE TITLE
Accessibility addon: Add severity badges to violation rows

### ISSUE THREAD

**Issue by domyen** [storybookjs/storybook#31417]
### Discussed in https://github.com/storybookjs/storybook/discussions/31413

<div type='discussions-op-text'>

<sup>Originally posted by **MichaelArestad** May  7, 2025</sup>
### Is your feature request related to a problem? Please describe.

It can be difficult to prioritize which accessibility issues to fix first. This will help with that.

### Describe the solution you'd like

We should be adding severity badges to the violation rows. I would like to see them implemented like this:
<img width="1072" alt="image" src="https://github.com/user-attachments/assets/39d58c8c-dd75-4698-88f6-941a48cb87f5" />

[Source file is here](https://www.figma.com/design/76ZT7rCGI8AAoqs77SCsGr/Accessibility-in-Storybook?node-id=907-6583&t=4ZTpugKHnFGq82Yi-1)

### Details
1. Add the severity using a Badge component
2. Align the spacing so that in most instances, the count and severity line up with the actions in the subnav above it.

### Here are the possible severity states:
<img width="165" alt="image" src="https://github.com/user-attachments/assets/b974475b-8363-4a63-bd02-556f846e5a5a" />

### Are you able to assist to bring the feature to reality?

yes, I can

### Additional context

Only from the design side of things</div>

**Comment by domyen** [comment]
@MichaelArestad should we show the severity badges for the passed & inconclusive tabs too?

I can see two options:
1. Show badges on other tabs but make them greyed out so as to not draw undue attention. 
2. Not show badges on other tabs at all

**Comment by MichaelArestad** [comment]
@domyen Yep! I think we could go with option 1. That's what I would have suggested.

**Comment by ghengeveld** [comment]
@domyen @MichaelArestad I see the rule name is back as well, at least it is in those designs. Should I reintroduce it as part of this ticket?

<img width="390" alt="Image" src="https://github.com/user-attachments/assets/bcb2872a-3e5a-483b-bbe4-7a513aeebb3c" />
